tests/import_test.go

```diff
@@ -36,4 +40,18 @@ var _ = Describe(testSuiteName, func() {
 		Expect(framework.VerifyPVCIsEmpty(f, pvc)).To(BeTrue())
 		// Not deleting PVC as it will be removed with the NS removal.
 	})
+	It("Import pod status should be Fail on unavailable endpoint", func() {
+		pvc, err := f.CreatePVCFromDefinition(utils.NewPVCDefinition(
+			"no-import",
+			"1G",
+			map[string]string{controller.AnnEndpoint: invalidEndpoint},
+			nil))
+		Expect(err).ToNot(HaveOccurred())
+		// Wait a while to see if CDI puts anything in the PVC.
+		time.Sleep(waitForCDIImportTimeout)
```

Comment:
unconditional sleeps generally should be avoided. It's usually better to poll for a condition with a small sleep between each iteration. The advantages of this approach are 1) potential for a much shorter sleep time thus helping CI to run faster, and 2) better chance of sleeping long enough for the desired condition to occur.

Reply:
utils.WaitForPVCAnnotation in line 53 does the polling already.
Should we remove the sleep then? Do you see why it's there if we are polling anyway?

Reply:
I know your didn't change the func below but I reviewed it based on your reply above, and I think it should be refactored a bit due to a) idiomatic go, b) bug where nil pvc ptr can be referenced:
```
func WaitForPVCAnnotation(clientSet *kubernetes.Clientset, namespace string, pvc *k8sv1.PersistentVolumeClaim, annoKey string) (annoValue string, found bool, err error) {
	err = wait.PollImmediate(PVCPollInterval, PVCCreateTime, func() (bool, error) {
		pvc, err := FindPVC(clientSet, namespace, pvc.Name)
		if err != nil {
			return false, nil // keep trying
		}
		if annoValue, found = pvc.ObjectMeta.Annotations[annoKey] ; found {
			return true, nil
		}
		return false, err
	})
	return
}
```

Reply:
and, to answer your question, I don't see a reason why the `sleep` above is needed.

Reply:
ok